### ISSUE TITLE
Inconsistency between Object Literal and constructor in strictNullChecks mode

### ISSUE THREAD

**Issue by DJCordhose** [Microsoft/TypeScript#9773]
**TypeScript Version:**   (2.0.0-beta)

 "strictNullChecks": true

**Code**

```typescript
class Person {
    name: string;
    constructor() {
        // why no error?
    }
}

const olli: Person = new Person();

```

**Expected behavior:**

Should issue error like in this example:

```typescript
const daniel: Person = {
};
```

error:

```
error TS2322: Type '{}' is not assignable to type 'Person'.
  Property 'name' is missing in type '{}'.
```

**Actual behavior:** 

shows no error

**Comment by mvestergaard** [comment]
+1 this seems wrong and leads to inconsistencies. It should require me to set the property in the constructor if undefined is not allowed.

**Comment by nippur72** [comment]
@mvestergaard it seems this is a `wontfix`, see #8476
